core/src/main/java/tc/oc/pgm/fireworks/FireworkMatchModule.java

```diff
@@ -0,0 +1,258 @@
+package tc.oc.pgm.fireworks;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.bukkit.Bukkit;
+import org.bukkit.Color;
+import org.bukkit.FireworkEffect;
+import org.bukkit.FireworkEffect.Type;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Firework;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.meta.FireworkMeta;
+import tc.oc.pgm.Config;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchFinishEvent;
+import tc.oc.pgm.api.party.Competitor;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.controlpoint.events.ControllerChangeEvent;
+import tc.oc.pgm.core.CoreLeakEvent;
+import tc.oc.pgm.destroyable.DestroyableDestroyedEvent;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.flag.event.FlagCaptureEvent;
+import tc.oc.pgm.regions.Bounds;
+import tc.oc.pgm.regions.Region;
+import tc.oc.pgm.wool.PlayerWoolPlaceEvent;
+import tc.oc.util.bukkit.BukkitUtils;
+import tc.oc.util.bukkit.block.BlockVectors;
+
+@ListenerScope(MatchScope.LOADED)
+public class FireworkMatchModule implements MatchModule, Listener {
+
+  private final Match match;
+
+  private static final boolean POST_ENABLED = Config.Fireworks.postMatchEnabled();
+  private static final boolean GOALS_ENABLED = Config.Fireworks.goalsEnabled();
+
+  private static final int ROCKET_COUNT = 5; // Maximum rockets to launch at once, one per player
+  private static final int INITIAL_DELAY = 40; // Ticks before starting to launch rockets
+  private static final int FREQUENCY = 40; // Ticks between rocket launches
+  private static final int ITERATION_COUNT = 15; // Amount of times rockets are launched
+  private static final int ROCKET_POWER =
+      2; // Power applied to rockets (how high they go), 1 = low, 2 = medium, 3 = high
+
+  private static final int WOOL_PLACE_COUNT = 6;
+  private static final double WOOL_PLACE_RADIUS = 2;
+
+  private static final int CORE_LEAK_COUNT = 8;
+  private static final double CORE_LEAK_RADIUS = 1.5;
+
+  private static final int DESTROYABLE_BREAK_COUNT = 4;
+  private static final double DESTROYABLE_BREAK_RADIUS = 3;
+
+  private static final int CONTROL_POINT_COUNT = 8;
+  private static final double CONTROL_POINT_RADIUS = 1;
+
+  private static final int FLAG_CAPTURE_COUNT = 6;
+  private static final double FLAG_CAPTURE_RADIUS = 1;
```

Comment:
TBH, if there are all just "magic" numbers, then let's skip the static variables.
Nice in theory, but there's way too many vars here.

Reply:
I'd much rather have them as variables than magic numbers. If we ever feel like changing them it's much easier to have it this way

Reply:
@Electroid I agree with @Pablete1234. It is rather excessive for magic numbers, but personally I feel it’s more organized and understandable for new contributors who may want to adjust them later on. 

If you dislike them all at the top of the file, I can move each respective pair above the event they’re related to like:

```java
...
private static final int CORE_LEAK_COUNT = 8;
private static final double CORE_LEAK_RADIUS = 1.5;

 @EventHandler(priority = EventPriority.MONITOR)
   public void onCoreLeak(final CoreLeakEvent event) {
     if (GOALS_ENABLED && event.getCore().isVisible()) {
       this.spawnFireworkDisplay(
           event.getMatch().getWorld(),
           event.getCore().getCasingRegion(),
           event.getCore().getColor(),
           CORE_LEAK_COUNT,
           CORE_LEAK_RADIUS,
           ROCKET_POWER);
     }
   }

```

Reply:
@applenick Go ahead and make that change, then ready to go